src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -458,42 +473,53 @@ private bool HandleError(C4Error error)
             return true;
         }
 
-        private void OnDocEnded(List<ReplicatedDocument> replications, bool pushing)
+        private void OnDocEnded(ConcurrentBag<ReplicatedDocument> replications, bool pushing)
         {
             if (_disposed) {
                 return;
             }
 
-            for(int i = 0; i < replications.Count; i++) {
+            while (!replications.IsEmpty) {
+                _conflictTasks.Add(Task.Run(() =>
+                {
+                    ReplicatedDocument replication;
+                    if (replications.TryTake(out replication)) {
+                        try {
+                            Config.Database.ResolveConflict(replication.Id, Config.ConflictResolver);
+                            replication = replication.ClearError();
+                        } catch (Exception e) {
+                            C4Error error;
+                            if(e.GetBaseException() is CouchbaseException) {
+                                var errCode = ((CouchbaseException)e).Error;
+                                error = new C4Error((C4ErrorCode)errCode);
+                            } else {
+                                error = new C4Error(C4ErrorCode.UnexpectedError);
+                            }
+                            replication = replication.AssignError(error);
```

Comment:
It would be good to preserve the original exception's message here.  The tests should be checking this.  Consider setting the error *and* the exception by altering the `ReplicatedDocument` class